hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/AbstractSerializationService.java

```diff
@@ -425,8 +427,11 @@ protected final boolean safeRegister(final Class type, final Serializer serializ
     }
 
     protected final boolean safeRegister(final Class type, final SerializerAdapter serializer) {
-        if (constantTypesMap.containsKey(type)) {
-            throw new IllegalArgumentException("[" + type + "] serializer cannot be overridden");
+        if (constantTypesMap.containsKey(type) && !allowOverrideDefaultSerializers) {
+            throw new IllegalArgumentException(
+                "[" + type + "] serializer cannot be overridden."
+                + " See hazelcast.serialization.allowOverrideDefaultSerializers"
```

Comment:
Since we will be removing the property, please do not forget to update this part as `SerializationConfig. setAllowOverrideDefaultSerializers`

Reply:
Message updated